### PR TITLE
[Bug](function) fix cut_ipv6 function error about modify the input column data

### PR DESCRIPTION
**Pull request by zhangstar333**
### What problem does this PR solve?
Problem Summary:
cut_ipv6

the ipv6 function maybe need reverse the input ipv6 data, but can't do it at the input data
could copy to another data area process it.

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/functions/function_ip.h [hunk 1]
```diff
@@ -1318,9 +1321,10 @@ class FunctionCutIPv6 : public IFunction {
                                 get_name());
             }
 
-            UInt8 bytes_to_cut_count = is_ipv4_mapped(address) ? bytes_to_cut_for_ipv4_count
-                                                               : bytes_to_cut_for_ipv6_count;
-            cut_address(address, pos, bytes_to_cut_count);
+            UInt8 bytes_to_cut_count = is_ipv4_mapped(ipv6_address_data)
+                                               ? bytes_to_cut_for_ipv4_count
+                                               : bytes_to_cut_for_ipv6_count;
+            cut_address(ipv6_address_data, pos, bytes_to_cut_count);
```
Comment (HappenLee) [inline be/src/vec/functions/function_ip.h:1327]: Add detail comment for the code. And consider we pass reference or value replace by ptr

**Comment by dataroaring** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**